### PR TITLE
Clean up docs and PHPDocs

### PR DESCRIPTION
**Pull request by spawnia**
- ~[ ] Added or updated tests~
- [x] Added or updated the [README.md](../README.md)
- [x] Detailed changes in the [CHANGELOG.md](../CHANGELOG.md) unreleased section

**Related Issue/Intent**

The Changelog should reflect relevent changes.

**Changes**

- Mention breaking dependency upgrade in Changelog
- Fix some PHPDocs and formatting

**Breaking changes**

No

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -19,9 +19,14 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 - Native attribute casting [#131](https://github.com/BenSampo/laravel-enum/pull/131)
 
+### Changed
+
+- Require Laravel 7.5 or higher
+- Require PHP 7.2.5 or higher
+
 ### Deprecated
 
-- Dropped support for legacy attribute casting in favour of native casting
```
Comment (spawnia) [inline CHANGELOG.md:24]: This sounds like the support is already dropped, which is not true.

File: src/Casts/EnumCast.php [hunk 2]
```diff
@@ -8,25 +8,19 @@
...
+    /** @var string */
     protected $enumClass;
 
     public function __construct(string $enumClass)
     {
         $this->enumClass = $enumClass;
     }
 
-    /**
-     * @inheritDoc
```
Comment (spawnia) [inline src/Casts/EnumCast.php:20]: Those only help when doing automatic docs generation. We do not use any such tooling.

**Comment by spawnia** [comment]
> May I ask the reason why you are adding FQCNs in the docblocks?

I think it is a good idea for the PHPDocs to be as descriptive as possible, a bit of extra verbosity does not hurt. The code itself does not become less readable as a result. It is also consistent with the way Laravel itself does it.

**Comment by SeinopSys** [comment]
@atymic Something worth noting is that because of https://github.com/barryvdh/laravel-ide-helper/issues/627 any non-FQCNs would appear incorrectly in the generated helper file, so if for no other reason then it's worth keeping them juts to avoid potential issues with this tool.

**Comment by BenSampo** [comment]
Thanks @spawnia 👍

**Pull request merged**